services/streammanager/googlepubsub/googlepubsubmanager.go

```diff
@@ -99,9 +94,8 @@ func Produce(jsonData json.RawMessage, producer interface{}, destConfig interfac
 			return 400, respStatus, responseMessage
 		}
 		var topic *pubsub.Topic
-		for _, s := range pbs.TopicMap {
-			splitString := strings.Split(s.String(), "/")
-			if splitString[3] == topicIdString {
+		for i, s := range pbs.TopicMap {
```

Comment:
You can access the topic as 
```
topic := pbs.TopicMap[topicIdString]
```
and get rid of this for loop.

Reply:
Also, make sure unnecessary things are not logged. like the logger.Info(s.String()) below.

Reply:
done. and removed.